### ISSUE TITLE
Permissions

### ISSUE THREAD

**Issue by nukeop** [Pauan/AutoScroll#18]
Why does this need the permission to "Read and change all your data on the websites you visit"?

**Comment by Pauan** [comment]
What that permission really means is, "the extension can run JavaScript code on all websites".

Chrome does not allow extensions to control the browser, so the **only** way to implement autoscroll is to add some JavaScript code to every website, and that JavaScript code adds in the autoscroll feature.

[Here is the code which specifies which JavaScript code will be run on all websites.](https://github.com/Pauan/AutoScroll/blob/master/src/manifest.json#L21-L26) As you can see, it runs the `data/defaults.js` and `data/AutoScroll.js` files on all websites.

[The `data/defaults.js` file just specifies the default options, nothing more.](https://github.com/Pauan/AutoScroll/blob/master/src/data/defaults.js)

[The `data/AutoScroll.js` file adds in the autoscroll feature.](https://github.com/Pauan/AutoScroll/blob/master/src/data/AutoScroll.js)

If you understand JavaScript, you can inspect those files yourself to see whether they do anything harmful or not.

The reason Chrome makes the permission warning sound so scary is because an extension *could* run some JavaScript code which analyzes and/or modifies the websites that you visit.

In other words, the extension *could* spy on you, or mess things up on the websites you visit.

Of course AutoScroll does not do anything like that, but it's good to be cautious.

**Comment by nukeop** [comment]
Thanks for clarifying, and thanks for sharing your code - upon inspection there's nothing suspicious in it.
